test/srtp_driver.c

```diff
@@ -160,19 +188,151 @@ srtp_master_key_t *test_keys[2] = {
 };
 // clang-format on
 
+bool use_srtp_not_in_place_io_api = false;
+
+void overrun_check_prepare(uint8_t *buffer, size_t offset, size_t buffer_len)
+{
+    memset(buffer + offset, 0xff, buffer_len - offset);
+}
+
+srtp_err_status_t call_srtp_protect2(srtp_ctx_t *ctx,
+                                     uint8_t *rtp,
+                                     size_t rtp_len,
+                                     size_t *srtp_len,
+                                     size_t mki_index)
+{
+    srtp_err_status_t status = srtp_err_status_fail;
+    if (use_srtp_not_in_place_io_api) {
+        uint8_t in_buf[4048];
+        if (rtp_len > sizeof(in_buf)) {
+            printf("rtp_len greater than in_buf");
+            exit(1);
+        }
+        memcpy(in_buf, rtp, rtp_len);
+        status = srtp_protect(ctx, in_buf, rtp_len, rtp, srtp_len, mki_index);
+    } else {
+        status = srtp_protect(ctx, rtp, rtp_len, rtp, srtp_len, mki_index);
+    }
+    return status;
+}
+
+srtp_err_status_t call_srtp_protect(srtp_ctx_t *ctx,
+                                    uint8_t *rtp,
+                                    size_t *rtp_len,
+                                    size_t mki_index)
+{
+    // an assumption
+    size_t srtp_len = *rtp_len + SRTP_MAX_TRAILER_LEN;
+    srtp_err_status_t status =
+        call_srtp_protect2(ctx, rtp, *rtp_len, &srtp_len, mki_index);
+    *rtp_len = srtp_len;
+    return status;
+}
+
+srtp_err_status_t call_srtp_unprotect2(srtp_ctx_t *ctx,
+                                       uint8_t *srtp,
+                                       size_t srtp_len,
+                                       size_t *rtp_len)
+{
+    srtp_err_status_t status = srtp_err_status_fail;
+    if (use_srtp_not_in_place_io_api) {
+        uint8_t in_buf[4048];
+        if (srtp_len > sizeof(in_buf)) {
+            printf("srtp_len greater than in_buf");
+            exit(1);
+        }
+        memcpy(in_buf, srtp, srtp_len);
+        status = srtp_unprotect(ctx, in_buf, srtp_len, srtp, rtp_len);
+    } else {
+        status = srtp_unprotect(ctx, srtp, srtp_len, srtp, rtp_len);
+    }
+    return status;
+}
+
+srtp_err_status_t call_srtp_unprotect(srtp_ctx_t *ctx,
+                                      uint8_t *srtp,
+                                      size_t *srtp_len)
+{
+    return call_srtp_unprotect2(ctx, srtp, *srtp_len, srtp_len);
```

Comment:
Why not just have `call_srtp_unprotect()` do the work of `call_srtp_unprotect2()`?

Reply:
Wanted to keep the existing call_srtp_unprotect() signature and try to remove the xxx2() functions in a followup cleanup commit, but i see now this one could have easily been merge already now.